src/Modal/Backdrop.d.ts

```diff
@@ -1,8 +1,13 @@
 import * as React from 'react';
 import { StandardProps } from '..';
 import { TransitionDuration } from '../internal/transition';
+import { FadeProps } from '../transitions/Fade';
 
-export interface BackdropProps extends StandardProps<{}, BackdropClassKey> {
+export interface BackdropProps
+  extends StandardProps<
+      React.HTMLAttributes<HTMLDivElement> & Partial<FadeProps>,
```

Comment:
Looking at the JS code, it looks like it should be

```ts
Omit<FadeProps, 'appear' | 'in' | 'timeout'>
```

The `React.HTMLAttributes<HTMLDivElement>` should be handled by `FadeProps` (really by `TransitionProps`, which it extends).

Reply:
Speaking of `TransitionProps`, why do we define our own type for it instead of using the one from `react-transition-group`?

Reply:
> The React.HTMLAttributes<HTMLDivElement> should be handled by FadeProps (really by TransitionProps, which it extends).

I don't think so. Fade is forwarding the properties to its children. The Backdrop is providing a div, the div properties are available. Unless there is a way to reflect this pattern in TypeScript?

> Speaking of TransitionProps, why do we define our own type for it instead of using the one from react-transition-group?

Good question. I don't know.

Reply:
Why do you want to remove `Omit<FadeProps, 'appear' | 'in' | 'timeout'>`? Users can provide the properties. They are available.

Reply:
> I don't think so. Fade is forwarding the properties to its children. The Backdrop is providing a div so the div properties are available. Unless there is a way to reflect this pattern in TypeScript?

`Fade` isn't forwarding properties to its children, only to the `Transition`, right?

```js
<Transition
  appear={appear}
  style={style}
  onEnter={this.handleEnter}
  onEntering={this.handleEntering}
  onExit={this.handleExit}
  {...other}
>
  {children}
</Transition>
```

> Why do you want to remove `Omit<FadeProps, 'appear' | 'in' | 'timeout'>`? Users can provide the properties. They are available. Am I missing something?

You're right, it should just be `Partial<FadeProps>`.

Reply:
> Fade isn't forwarding properties to its children, only to the Transition, right?

Yes, exactly. Should we merge the pull-request or do you think it should be modified?

Reply:
I think the `React.HTMLAttributes<HTMLDivElement> & ` should be removed.

Reply:
If we remove it, it doesn't solve @daniel-rabe issue. What's the alternative.

Reply:
The proper place to add `onTouch*` would be on `TransitionProps`, wouldn't it?

Reply:
> The proper place to add onTouch* would be on TransitionProps, wouldn't it?

I don't think so. `TransitionProps` is only responsible for the property the transition component is handling. The `onTouc*` methods are handled by its children. It's the Backdrop component that control the children element, not the Fade nor the Transition.

Reply:
Maybe I’m being dense, but I don’t see where props are being forwarded to the children?

Reply:
https://github.com/reactjs/react-transition-group/blob/5007303e729a74be66a21c3e2205e4916821524b/src/Transition.js#L343

Reply:
Ah, of course! Thanks for bearing with me.

Reply:
:) it would be great to move to the react-transition-group TypeScript definition at some point.

Reply:
> :) it would be great to move to the react-transition-group TypeScript definition at some point.

Made an issue for this: #10042